### PR TITLE
Documentation (and other) updates for 3.29.0

### PR DESCRIPTION
**Pull request by absurdfarce**

### CODE REVIEW

File: tests/unit/test_cluster.py [hunk 1]
```diff
@@ -88,8 +89,9 @@ class ClusterTest(unittest.TestCase):
 
     def test_tuple_for_contact_points(self):
         cluster = Cluster(contact_points=[('localhost', 9045), ('127.0.0.2', 9046), '127.0.0.3'], port=9999)
+        localhost_addr = set([addr[0] for addr in [t for (_,_,_,_,t) in socket.getaddrinfo("localhost",80)]])
         for cp in cluster.endpoints_resolved:
-            if cp.address in ('::1', '127.0.0.1'):
+            if cp.address in localhost_addr:
```
Comment (absurdfarce) [inline tests/unit/test_cluster.py:94]: Changes in this file intended to address PYTHON-1373

File: CHANGELOG.rst [hunk 2]
```diff
@@ -1,3 +1,19 @@
+3.29.0
+======
+December 19, 2023
+
+Features
+--------
+* Add support for Python 3.9 through 3.12, drop support for 3.7 (PYTHON-1283)
+* Removal of dependency on six module (PR 1172)
+* Raise explicit exception when we try to deserialize a vector with a subtype that isn’t constant size (PYTHON-1371)
```
Comment (emeliawilkinson24) [inline CHANGELOG.rst:9]:
```suggestion
* Raise explicit exception when deserializing a vector with a subtype that isn’t a constant size (PYTHON-1371)
```

File: CHANGELOG.rst [hunk 3]
```diff
@@ -1,3 +1,19 @@
...
+
+Others
+------
+* Remove outdated Python pre-3.7 references (PR 1186)
+* Removed backup(.bak) files (PR 1185)
```
Comment (emeliawilkinson24) [inline CHANGELOG.rst:14]:
```suggestion
* Remove backup(.bak) files (PR 1185)
```

File: README-dev.rst [hunk 4]
```diff
@@ -176,7 +176,7 @@ Use tee to capture logs and see them on your terminal::
 
 Testing Multiple Python Versions
 --------------------------------
-If you want to test all of python 3.7, 3.8, and pypy, use tox (this is what
+If you want to test all of python 3.8 through 3.12 (and pypy) use tox (this is what
```
Comment (emeliawilkinson24) [inline README-dev.rst:179]:
```suggestion
Use tox to test all of Python 3.8 through 3.12 and pypy (this is what
```

**Review by emeliawilkinson24 — APPROVED** [review]
Optional changes, LGTM!

**Pull request merged**